presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -293,6 +295,42 @@ public synchronized void renameDatabase(String source, String target)
         setExclusive((delegate, hdfsEnvironment) -> delegate.renameDatabase(source, target));
     }
 
+    public synchronized void setTableStatistics(Table table, PartitionStatistics tableStatistics)
+    {
+        setExclusive((delegate, hdfsEnvironment) -> delegate.updateTableStatistics(table.getDatabaseName(), table.getTableName(), statistics -> tableStatistics));
+    }
+
+    public synchronized void setPartitionStatistics(Table table, Map<List<String>, PartitionStatistics> partitionStatisticsMap)
+    {
+        setExclusive((delegate, hdfsEnvironment) ->
```

Comment:
Yes

Reply:
Sometimes transactions are implicit (eg in JDBC) and users may end up running transactionally unintentionally. 

If this is not easy to fix, let's leave this this way.

Reply:
I see. That makes sense. Although all the other DDL operations (and many DML) are not allowed to be run in the same transaction.

Reply:
This requires us to treat ANALYZE as a table action in the `SemiTransactionalHiveMetaStore` and I think it can be in a separate PR. I will leave a TODO.